src/Milky/Checker.cs

```diff
@@ -249,24 +249,19 @@ private void OutputCombo(Combo combo, CheckResult checkResult)
 
                     if (_outputSettings.SpecialColors.Count > 0 && checkResult.Captures != null)
                     {
-                        foreach (KeyValuePair<ConsoleColor, KeyValuePair<string, Func<object, bool>>> keyValuePair in _outputSettings.SpecialColors)
+                        foreach (KeyValuePair<ConsoleColor, KeyValuePair<string, Predicate<object>>> specialColor in _outputSettings.SpecialColors)
                         {
-                            object capturedObject = null;
-
-                            if (checkResult.Captures.TryGetValue(keyValuePair.Value.Key, out capturedObject))
+                            foreach (KeyValuePair<string, object> capture in checkResult.Captures.Where(x => (x.Key == specialColor.Value.Key && specialColor.Value.Value.Invoke(x.Value))))
```

Comment:
Actually just return it to how it was before, but with the `break` added.

Also, you can remove the `object capturedObject = null;` line and do `out var capturedObject` right away.